webapp/src/errorList.tsx

```diff
@@ -46,8 +48,12 @@ export class ErrorList extends React.Component<ErrorListProps, ErrorListState> {
             {!isCollapsed && collapseButton}
         </div>
 
+        const createOnErrorMessageClick = (e: pxtc.KsDiagnostic) => () => this.onErrorMessageClick(e.line + 1, e.column + 1)
+
         const errorListContent = (errors).map(e =>
-            <div key={errorKey(e)} className="errorMessage">{`${e.messageText} ${lf("at line {0}", e.line + 1)}`}</div>)
+            <div key={errorKey(e)} className="errorMessage" role="button" onClick={createOnErrorMessageClick(e)}>
```

Comment:
This change suppresses the error, but the actual cause is still there -- it'll keep recreating the function each time render is called. It may not be a big deal for a single element, but it's best to avoid that as it adds up.

Here's an example of a workaround for something similar: https://github.com/microsoft/pxt/blob/master/webapp/src/filelist.tsx#L387 - that stores the file, but you could store line / col as props on the message div and reference them from that

Reply:
Hmm, we could do a separate component per message which basically gives us a way to store some data on the local element.

I feel like in this case it should be fine.. the anonymous functions we're creating are just closing over a couple numbers. It's not obvious to me another approach would use fewer resource

Reply:
That's true almost any time this shows up though - you typically are only passing a param or two, so it's only ever an anonymous function wrapping a few references - the main cost is in extra rerenders for every item when the fn is considered not equal to the previous one each time the errorlist rerenders.

That won't be a noticeable cost because it's a smaller component, but it's easy enough to break out into an `ErrorListItem` component (and that helps keep `ErrorList.render` nice and tidy)

(This is a rather small thing that can be done separately though)

Reply:
i think you could add "data-row=" and "data-col=" attributes onto the div and use e.target to grab them in the event listener

Reply:
This is a micro-optimization and we can do another PR for this kind of cleanup. Push this fix out into another PR.